packages/database/src/migrations/20210923011806-LESMISEnrolmentRateMapOverlays-modifies-data.js

```diff
@@ -0,0 +1,235 @@
+            aggregationEntityType: 'requested',
+          },
+        },
+      ],
+    },
+    transform: [
+      {
+        transform: 'updateColumns',
+        insert: {
+          organisationUnitCode: '=$organisationUnit',
+          value: '=divide($value, 100)',
+        },
+        exclude: ['organisationUnit', 'dataElement', 'period'],
+      },
+    ],
+  },
+});
+
+const PERMISSION_GROUP = 'LESMIS Public';
+
+const getMapOverlay = (name, reportCode) => ({
+  id: reportCode,
+  name,
+  userGroup: PERMISSION_GROUP,
+  dataElementCode: 'value',
+  isDataRegional: true,
+  measureBuilder: 'useReportServer',
+  measureBuilderConfig: {
+    dataSourceType: 'custom',
+    reportCode,
+  },
+  presentationOptions: {
+    scaleType: 'performance',
+    displayType: 'shaded-spectrum',
+    measureLevel: reportCode.includes('Province') ? 'District' : 'SubDistrict',
+    valueType: 'percentage',
+    scaleBounds: {
+      left: {
+        max: 'auto',
+      },
+      right: {
+        min: 'auto',
+      },
+    },
+    periodGranularity: 'one_year_at_a_time',
+  },
+  countryCodes: '{"LA"}',
+  projectCodes: '{laos_schools}',
+
+  legacy: false,
+  report_code: reportCode,
+});
+
+const addMapOverlayGroup = async (db, parentCode, { name, code }) => {
+  const parentId = await codeToId(db, 'map_overlay_group', parentCode);
+
+  const overlayGroupId = generateId();
+  await insertObject(db, 'map_overlay_group', { id: overlayGroupId, name, code });
+
+  return insertObject(db, 'map_overlay_group_relation', {
+    id: generateId(),
+    map_overlay_group_id: parentId,
+    child_id: overlayGroupId,
+    child_type: 'mapOverlayGroup',
+  });
+};
+
+const addMapOverlay = async (db, parentCode, config) => {
+  const { reportCode, dataElements, name, sortOrder } = config;
+  const report = getReport(reportCode, dataElements);
+  const mapOverlay = getMapOverlay(name, reportCode);
+  const permissionGroupId = await nameToId(db, 'permission_group', PERMISSION_GROUP);
+  await insertObject(db, 'report', {
+    id: generateId(),
+    permission_group_id: permissionGroupId,
+    ...report,
+  });
+  await insertObject(db, 'mapOverlay', mapOverlay);
+
+  const mapOverlayGroupId = await codeToId(db, 'map_overlay_group', parentCode);
+  return insertObject(db, 'map_overlay_group_relation', {
+    id: generateId(),
+    map_overlay_group_id: mapOverlayGroupId,
+    child_id: mapOverlay.id,
+    child_type: 'mapOverlay',
+    sort_order: sortOrder,
+  });
+};
+
+const OVERLAY_CODE = 'LESMIS_enrolment_ece';
+
+const OVERLAY_GROUPS = [
+  {
+    parentCode: 'Root',
+    children: [
+      {
+        name: 'ECE/Pre-Primary Enrolment',
+        code: `${OVERLAY_CODE}_Group`,
+      },
+    ],
+  },
+  {
+    parentCode: `${OVERLAY_CODE}_Group`,
+    children: [
+      {
+        name: 'District Level',
+        code: `${OVERLAY_CODE}_District_Group`,
+      },
+    ],
+  },
```

Comment:
This seems a bit redundant if we're not adding province level reports. Perhaps there will be for other map overlays?

Reply:
I don't think there's any province level reports for this particular category (ECE/Pre-Primary Enrolment), but I think it's better to keep it so that nested structure is consistent between all the different categories.

Reply:
Sure!